solidity/contracts/PhasedEscrow.sol

```diff
@@ -70,26 +70,127 @@ contract PhasedEscrow is Ownable {
     }
 }
 
-interface ICurveRewards {
-    function notifyRewardAmount(uint256 amount) external;
+/// @title BatchedPhasedEscrow
+/// @notice A token holder contract allowing contract owner to approve a set of
+///         beneficiaries of tokens held by the contract, to appoint a separate
+///         drawee role, and allowing that drawee to withdraw tokens to approved
+///         beneficiaries in phases.
+contract BatchedPhasedEscrow is Ownable {
+    using SafeERC20 for IERC20;
+
+    event BeneficiaryApproved(address beneficiary);
+    event TokensWithdrawn(address beneficiary, uint256 amount);
+    event DraweeRoleTransferred(address oldDrawee, address newDrawee);
+
+    IERC20 public token;
+    address public drawee;
+    mapping(address => bool) private approvedBeneficiaries;
```

Comment:
do we need to make it private? If it is set to public, then I think we can remove external function `isBeneficiaryApproved()` and just call `approvedBeneficiaries(address)`

Reply:
Yeah, I was thinking about it but `isBeneficiaryApproved(beneficiary)` reads better to me than `approvedBeneficiaries(address)` - that's why I decided to go with a separate function and making this mapping `private`.

Reply:
Ok